compiler/src/iree/compiler/GlobalOptimization/ExpandVectors.cpp

```diff
@@ -28,7 +28,7 @@ struct ExpandVectors
   LogicalResult matchAndRewrite(linalg::ContractionOpInterface op,
                                 PatternRewriter &rewriter) const override {
     auto linalgOp = dyn_cast<linalg::LinalgOp>(op.getOperation());
-    if (!linalgOp.hasTensorSemantics()) {
+    if (!linalgOp || !linalgOp.hasTensorSemantics()) {
```

Comment:
I thought all the ops LinalgOp is a super set of ContractionOpInterface? All the ContractionOpInterface ops should already implement LinalgOp details?

Reply:
Oh, I see. I just saw we use `dyn_cast` and it is possible to return null

Reply:
We should use `cast` instead